### PR TITLE
Mark motion events nullable.

### PR DESCRIPTION
**Pull request by yunikkk**
### Summary of changes

In https://github.com/mapbox/mapbox-maps-android/pull/1645 gesture library was bumped but  MotionEvents need to be marked nullable to fix the original issue #1019. 

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Mapbox welcomes participation and contributions from everyone.
 * Fix broken view annotation positioning when marking them `View.INVISIBLE` and then making `View.VISIBLE`. ([1616](https://github.com/mapbox/mapbox-maps-android/pull/1616))
 * Snap puck to north if puckBearingEnabled is set to false. ([1635] (https://github.com/mapbox/mapbox-maps-android/pull/1635))
 * Preserve cached properties if applied to the layer before during the `Style#getLayer` call. ([1622](https://github.com/mapbox/mapbox-maps-android/pull/1622))
-* Fix a `NullPointerException` in `StandardGestureListener`, where `MotionEvent` should be nullable. ([1645] (https://github.com/mapbox/mapbox-maps-android/pull/1645))
+* Fix a `NullPointerException` in `StandardGestureListener`, where `MotionEvent` should be nullable. ([1645] (https://github.com/mapbox/mapbox-maps-android/pull/1645)), ([1677] (https://github.com/mapbox/mapbox-maps-android/pull/1677))
```
Comment (jush) [inline CHANGELOG.md:14]: Remove empty space between PR ids and links:
```suggestion
* Fix a `NullPointerException` in `StandardGestureListener`, where `MotionEvent` should be nullable. ([1645](https://github.com/mapbox/mapbox-maps-android/pull/1645)), ([1677](https://github.com/mapbox/mapbox-maps-android/pull/1677))
```

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 2]
```diff
@@ -429,28 +429,34 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
...
      * Called when an on double tap gesture was detected.
      */
-    override fun onDoubleTapEvent(motionEvent: MotionEvent): Boolean {
+    override fun onDoubleTapEvent(motionEvent: MotionEvent?): Boolean {
+      if (motionEvent == null) {
+        return false
+      }
       return if (handleDoubleTapEvent(motionEvent, doubleTapMovementThreshold)) {
         true
       } else super.onDoubleTapEvent(motionEvent)
```
Comment (jush) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:462]: Should super be called even if it's null? If so, maybe handle the null inside `handleDoubleTapEvent(...)`

Comment (yunikkk) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:462]: Hard to say, those cases should not really happen at all - haven't been able to reproduce locally even one single time. Seems it's either buggy implementation of some vendors or Android version.
Also, we've only had reports about such cases for the onFling gesture.
That said, I'd probably left it as is, ignoring such gestures completely.

Comment (yunikkk) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:462]: Also, `super` here is just simple instance of listener, it does nothing https://github.com/mapbox/mapbox-gestures-android/blob/master/library/src/main/java/com/mapbox/android/gestures/StandardGestureDetector.java#L98

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 3]
```diff
@@ -459,19 +465,27 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
...
      */
     override fun onFling(
-      e1: MotionEvent,
-      e2: MotionEvent,
+      e1: MotionEvent?,
+      e2: MotionEvent?,
       velocityX: Float,
       velocityY: Float
     ): Boolean {
+      if (e1 == null || e2 == null) {
```
[the remark this answers is not in the document]
Comment (paulrenn67) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:485]: Maybe we can remove `e1` from `handleFlingEvent` if it is not needed.

[the remark this answers is not in the document]
Comment (yunikkk) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:485]: I'd say it's similar to the https://github.com/mapbox/mapbox-maps-android/pull/1677#discussion_r975104722 - previously we've crashed, happens rarely (and should not happen at all) - so I'd just ignored such gestures as a safest way to fix and not break anything.

**Review by jush — APPROVED** [review]

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (yunikkk) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:485]: > Maybe we can remove `e1` from `handleFlingEvent` if it is not needed.

Okay, removed

**Review by kiryldz — APPROVED** [review]

**Pull request merged**